### DISCUSSION TITLE
portname with midi-channels > 16

### DISCUSSION THREAD

**Discussion by yukulele** [FluidSynth/fluidsynth#1008] [Q&A]
When i start FluidSynth with:
```bash
fluidsynth ./sf2/FluidR3_GM.sf2
```
FluidSynth create a midi port with a name like `Synth input port (75216:0)` but the number change on each run.
So I use `--portname` to force a fixed name:
```bash
fluidsynth --portname "my midi synth" ./sf2/FluidR3_GM.sf2
```
I get `my midi synth` as port name

Now I want more than 16 channels so i use `--midi-channels`.
If I use it without port name:
```bash
fluidsynth --midi-channels 32 ./sf2/FluidR3_GM.sf2
```
I get two distinct random port name like:
```
Synth input port (75216:0)
Synth input port (75216:1)
```
If I add `--portname`:
```bash
fluidsynth --midi-channels 32 --portname 'my midi synth' ./sf2/FluidR3_GM.sf2
```
I get two identical port name:
```
my midi synth
my midi synth
```

how can I get several midi ports with static distinct port name ?

**Comment by derselbst** [comment]
> how can I get several midi ports with static distinct port name ?

Currently, you can't. If a custom portname is given, it will be used for all ALSA ports.

https://github.com/FluidSynth/fluidsynth/blob/8eaa91bf3b8297ec71519f8c296b23ff3891622f/src/drivers/fluid_alsa.c#L1119-L1126

Changing it should be straight forward. Provided that you don't need the ability to customize the "distinctiveness" of the portnames.

I would suggest to simply use:

```
my midi synth (12345:0)
my midi synth (12345:1)
```

Let me know your preference. I think I'll have time for that over Christmas holidays.

**Reply by yukulele** [reply]
> I would suggest to simply use:
> 
> ```
> my midi synth (12345:0)
> my midi synth (12345:1)
> ```

This would be perfect, but the port name shouldn't have any randomness, as this would mean reconfiguring the software's midi outputs each time it's launched. 

Something like this would be preferable:
```
my midi synth (0)
my midi synth (1)
```

**Reply by derselbst** [reply]
The fact that it took you 2 years and 10 months to reply makes me consider this issue as an absolute low-priority topic.

**Reply by yukulele** [reply]
Sorry, I had totally missed your answer. :sob: 

Actually, I still occasionally face this problem when working on projects requiring more than 16 channels.
